aries_vcx/src/core/profile/modular_libs_profile.rs

```diff
@@ -21,10 +22,16 @@ pub struct ModularLibsProfile {
 
 impl ModularLibsProfile {
     pub fn new(wallet: Arc<dyn BaseWallet>, ledger_pool_config: LedgerPoolConfig) -> VcxResult<Self> {
-        let ledger_pool = Arc::new(IndyVdrLedgerPool::new(ledger_pool_config)?);
-        let submitter = Arc::new(IndyVdrSubmitter::new(ledger_pool));
-        let ledger = Arc::new(IndyVdrLedger::new(Arc::clone(&wallet), submitter));
         let anoncreds = Arc::new(IndyCredxAnonCreds::new(Arc::clone(&wallet)));
+        let ledger_pool = Arc::new(IndyVdrLedgerPool::new(ledger_pool_config)?);
+        let request_submitter = Arc::new(IndyVdrSubmitter::new(ledger_pool));
+        let response_parser = Arc::new(ResponseParser::new());
+        let config = IndyVdrLedgerConfig {
```

Comment:
Maybe more related to PR - not sure if `request_submitter`, `response_parser` need to be Arc. I think `IndyVdrLedger` is going to be the only one to use them

Reply:
>  instead of putting things in this config, we could just tweak ndyVdrLedger::new signature and pass it directly. It's not like we have 10 optional things in the "config", all these things are pretty esential.

I put them into config very deliberately. I am planning to add more arguments and I am strongly convinced it is way easier to manage a single config object than many separate arguments.